### ISSUE TITLE
fix: autogalaxy_workspace_test multi_dataset delaunay 1805s release timeout

### ISSUE THREAD

**Issue by Jammy2211** [PyAutoLabs/autogalaxy_workspace_test#118]
## Heart RED context

This issue is a **corrective fix** for a live PyAutoHeart RED reason. It clears, in part, this reason (quoted verbatim):

> release validation FAILED (stage integrate)

This task owns the `autogalaxy_test scripts/multi_dataset/jax_likelihood/delaunay.py` TIMEOUT (1805 s) in integrate run 34094964905. It is the **only** one of the six corrective tasks whose failure appears on the integrate channel alone — it is not in the workspace-smoke reason. **Sibling RED reasons remain until every corrective PR lands** — the reason string above is not cleared by this PR alone; four sibling corrective tasks own the other integrate entries, and a fifth owns the separate workspace-validation reason.

Per the **corrective-PR exception** in `PyAutoBrain/AUTONOMY.md`, work on this task under a RED Heart requires a **human authorisation comment on this issue** before any commit, push or PR. Planning, reproduction and read-only investigation may proceed; nothing may be written to a branch until that comment exists.

**Mind prompt:** `PyAutoMind/active/multi_dataset_jax_likelihood_delaunay_release_profile_timeout.md` (was `draft/bug/autogalaxy_workspace_test/multi_dataset_jax_likelihood_delaunay_release_profile_timeout.md`)

## Overview

`scripts/multi_dataset/jax_likelihood/delaunay.py` times out at the 1800 s release cap on the integrate channel while passing in 14.99 s under the smoke profile. The prompt reads this as an env-profile split and asks which release setting triggers it. Investigation below shows there is very likely **no such setting**: the script's resolved env is effectively identical under both profiles, and the evidence instead matches the known intermittent `xla-cpu-eigen-pool-deadlock` family, of which a sibling script in this very repo is already the quarantined fourth member. The first step is therefore to falsify the prompt's premise before doing anything else.

## Verified findings

*(These challenge the prompt's premise and must not be re-derived from scratch.)*

- **The script's env grid is effectively IDENTICAL under both profiles.** It declares `ENV: jax full_datasets` (line 29), and `PyAutoHands/autohands/env_config.py:94` maps `full_datasets` → unset `PYAUTO_SMALL_DATASETS`; the `jax` token unsets `PYAUTO_DISABLE_JAX`. Diffing the two profiles with comments stripped, the only remaining deltas are `PYAUTO_TEST_MODE` 2 vs 0 and `PYAUTO_FAST_PLOTS` 1 vs 0 (plus `derive_jax_markers` and the cap). **The script runs no search and draws no plots** — it builds a `FactorGraphModel`, calls `fitness._vmap` twice and a `jax.jit`-wrapped `log_likelihood_function` once (lines 195-260). So neither remaining var reaches it. **The dataset, the over-sample grid, the jit config and the XLA flags are the same in both runs.** The prompt's hypothesis ("identify which release-profile env setting triggers it") has, on this evidence, **no candidate setting to find**.
- **What the evidence does match is the known `xla-cpu-eigen-pool-deadlock` family.** `autogalaxy_workspace_test/config/build/no_run.yaml:36` quarantines the **sibling** `multi_dataset/jax_likelihood/rectangular.py` for *exactly* this signature: "hung 1805s at the release cap … with `XLA_FLAGS=--xla_cpu_multi_thread_eigen=false` in force — same execution-hang signature as the family below (epic xla-cpu-eigen-pool-deadlock) … intermittent since — it passed integrate runs 33230028707 and 33255903385 the same day". The comment names it the **fourth** family member, after `autolens_workspace_test`'s `delaunay.py`, `shared_preloads.py` and `delaunay_mge.py` — all *multi_dataset composite vmap* scripts, exactly what this one is. `eec09d6`'s own commit message says the multi-dataset composite vmap structure is *"the Eigen-pool reproducer"* and was deliberately preserved.
- The family is **intermittent and bimodal** (the profile comment records 4/6 HANG vs 6/6 PASS ABAB data, Fisher p≈3e-6). A single smoke pass at 14.99 s against two release failures is fully consistent with intermittency and is **not** evidence of an env-profile split. The prompt's framing rests on n=1 per arm.
- `eec09d6` changed exactly one line in this script: `pixel_scales = 0.1` → `0.2`. Note the resulting mesh/mask ratio: `pixels = 500`, `edge_pixels_total = 30`, `mask_radius = 3.0`, so at 0.2"/px the mask holds ≈707 pixels against a 500-pixel mesh — ratio 0.71, near-degenerate. That is a *plausible* second mechanism (a near-singular solve compiling or executing pathologically) and is worth measuring, but it does not by itself explain a profile split either.
- Local jax is 0.10.2; CI ran 0.11.1.

## Plan

1. First, falsify the prompt's premise: dump the fully-resolved env this script receives under each profile and show whether they differ at all.
2. If they do not differ, re-classify: run the script N times under the release profile to measure the hang rate, and test it against the known Eigen-pool signature (compile finishes in ~12-18 s, then materialisation never returns).
3. Only if the env genuinely differs, bisect the differing setting.
4. Separately measure the 500-mesh / 707-mask ratio as a candidate mechanism.
5. Route the outcome: script/model fix if the ratio is the cause; family quarantine + epic escalation if it is the Eigen-pool hang. Do not raise the cap either way.

<details>
<summary>Detailed implementation plan</summary>

### Work Classification

**Workspace** (`autogalaxy_workspace_test`), with a live possibility that it is really an infrastructure/epic item.

### Affected Repositories

- autogalaxy_workspace_test (primary)

### Branch Survey

| Repository | Current Branch | Dirty? |
|-----------|---------------|--------|
| ./autogalaxy_workspace_test | main…origin/main | clean, no local feature branches |

No worktree claim.

**Suggested branch:** `feature/agwt-multi-delaunay-release-timeout`
**Worktree root:** `~/Code/PyAutoLabs-wt/agwt-multi-delaunay-release-timeout/` (created later by `/start_workspace`)

### Implementation Steps

- **Step 1 (cheap, decisive).** Instrument `PyAutoHands/autohands/env_config.py`'s resolver (or a small standalone call to `load_env_config` + the per-script resolution) to print the final env dict for `scripts/multi_dataset/jax_likelihood/delaunay.py` under `config/build/profile_smoke.yaml` and `config/build/profile_release.yaml`, and diff them. My reading of the config says the diff will be `{PYAUTO_TEST_MODE: 2→0, PYAUTO_FAST_PLOTS: 1→0}` and nothing else. **Confirm this before any other work** — it decides the whole task.
- **Step 2 (if the env is equivalent).** Run under the release profile with a heartbeat, 5-10 repeats, cold JAX cache, recording (a) time to compile complete, (b) time to `block_until_ready`. The family signature is compile ≈12-18 s then no return. If reproduced: the finding is "this is the fifth member of the `xla-cpu-eigen-pool-deadlock` family, not an env-profile split". Deliverable becomes a `no_run.yaml` quarantine entry written in the same forensic style as line 36's, cross-referenced to the epic, plus a corrected prompt/issue — and the human decides whether that clears the Heart reason or whether the epic must be worked. Say plainly that quarantining is a containment, not a fix.
- **Step 3 (mechanism probe, run regardless).** Reduce `pixels` from 500 toward ~350 (ratio ~0.5, matching the sibling `delaunay_mge.py` which passes at ~28 s under both profiles) and measure. If the hang rate collapses, that is a real, shippable model fix in the script — and the better outcome, since it removes the reproducer rather than hiding it. Compare directly against `delaunay_mge.py`'s mesh/mask ratio to justify the number.
- **Step 4.** Whichever lands, re-pin nothing (the script's asserts are `rtol=1e-2` parity checks, not absolute literals) and re-run 3× under the release profile to demonstrate stability, not a single lucky pass.
- Trade-off to state: the prompt's `Unattended: ready` flag is questionable given the likely reclassification. Recommend downgrading it.

### Key files

- `autogalaxy_workspace_test/scripts/multi_dataset/jax_likelihood/delaunay.py` — the failing script (`ENV:` line 29; `pixels`/`mask_radius`; the vmap + jit calls at lines 195-260)
- `autogalaxy_workspace_test/config/build/no_run.yaml:36` — the sibling `rectangular.py` quarantine, the template for a containment entry
- `PyAutoHands/autohands/env_config.py:94` — the `full_datasets` token mapping (the premise falsification)

</details>

## Repro

```
cd autogalaxy_workspace_test
PYTHONPATH=../PyAutoNerves:../PyAutoArray:../PyAutoFit:../PyAutoGalaxy:../PyAutoHands \
python3 ../PyAutoHands/autohands/run_python.py autogalaxy_test \
    scripts/multi_dataset/jax_likelihood \
    --env-config config/build/profile_release.yaml
```

(and the same with `profile_smoke.yaml`). Local wall-clock will not match CI; the *hang vs completes* verdict is what transfers, not the seconds.

## Witness

From the prompt: completes under `profile_release.yaml` in under 300 s with the triggering setting named.

**Flag:** if step 1 shows no differing setting, this witness is **unsatisfiable as written** and must be renegotiated with the human before the task ships — e.g. "completes under `profile_release.yaml` in under 300 s on 3 of 3 consecutive cold-cache runs".

## Risks

The prompt's causal story is probably wrong and the task may reclassify into a known open epic, which the human should decide on rather than an autonomous run. Intermittency means any "it passes now" result needs repeats to mean anything. Local jax 0.10.2 vs CI 0.11.1.

## Original Prompt

<details>
<summary>Click to expand starting prompt</summary>

```markdown
# autogalaxy_workspace_test multi_dataset/jax_likelihood/delaunay.py passes in 15 s under the smoke profile but times out at 1805 s under the release profile

Type: bug
Target: autogalaxy_workspace_test
Repos:
- autogalaxy_workspace_test
Difficulty: medium
Autonomy: supervised
Priority: medium
Status: formalised
Filed: 2026-09-07
Witness: `scripts/multi_dataset/jax_likelihood/delaunay.py` completes under `profile_release.yaml` in under 300 s (it was 1805 s TIMEOUT on 2026-09-07 and 14.99 s under `profile_smoke.yaml`), with the triggering release-profile setting named in the PR.
Unattended: ready

On the Release Integrate rehearsal (PyAutoHeart run 34094964905, and the preceding
34018429178 on 2026-09-06) the `autogalaxy_workspace_test` leg fails on:

```
scripts/multi_dataset/jax_likelihood/delaunay.py   TIMEOUT (1805s)   [profile_release.yaml, cap 1800]
```

Same run, same script under the smoke profile: passed in 14.99 s. The sibling
`delaunay_mge.py` passes at ~28 s under both profiles. The script was touched by the
2026-09-06 smoke-dataset rebuild commit `eec09d6`.

A 15 s -> 1805 s split between env profiles on one script, with its MGE sibling unaffected,
looks like a JAX compile explosion under the release env grid (precedent: a non-uniform
over-sample map tripled jit compile time and pushed SLaM simultaneous.py past its 1800 s
cap). Reproduce locally under `profile_release.yaml`, profile compile vs execute time,
identify which release-profile env setting triggers it (over-sample grid, dataset
resolution, jit config), and fix either the script or the library path it exposes. Do
not resolve by raising the cap.
```

</details>

**Comment by Jammy2211** [comment]
**Corrective-PR authorisation (PyAutoBrain/AUTONOMY.md, "Corrective-PR exception for Heart RED")**

Human authorisation given live in the Claude Code session on 2026-09-07, in the human's own words:

> authorise corrective PRs on #296, #297, #298, #299, autolens_workspace#537 and autogalaxy_workspace_test#118 against the two RED reasons

Heart RED reason(s) this issue's PR repairs, verbatim from `pyauto-heart readiness` at authorisation time:

- `release validation FAILED (stage integrate)`

Scope: commit, push and one pending-release PR whose diff maps to the reason(s) above. Not authorised: merge, issue close, release. Sibling RED reasons remain until every corrective PR lands; Heart stays RED and release stays blocked until then.

Session: https://claude.ai/code/session_01UUjCh13UJ62sgM3wN6XNuZ

**Comment by Jammy2211** [comment]
Corrective PR opened: https://github.com/PyAutoLabs/autogalaxy_workspace_test/pull/119 (pending-release; merge is a separate human act via /prm).

**Comment by Jammy2211** [comment]
## Shipped — 2026-09-07

### PRs
- https://github.com/PyAutoLabs/autogalaxy_workspace_test/pull/119 — merged (f7c9f6d3)

### Summary
No release-profile setting differs for this script; both CI hangs park in JAX execution under the vmap re-trace, the xla-cpu-eigen-pool-deadlock family. Shipped as a forensic `NEEDS_FIX` quarantine in `config/build/no_run.yaml` — containment, not a fix; remove when the epic lands. Library-side lever filed as `draft/refactor/autofit/fitness_vmap_outer_jit_halves_eigen_pool_exposure.md`. Corrective PR under the human-authorised Heart RED exception.
